llama-index-core/llama_index/core/instrumentation/span/__init__.py

```diff
@@ -1,5 +1,11 @@
+from contextvars import ContextVar
+from typing import Optional
+
 from llama_index.core.instrumentation.span.base import BaseSpan
 from llama_index.core.instrumentation.span.simple import SimpleSpan
 
+# ContextVar for managing active spans
+active_span_id: ContextVar[Optional[str]] = ContextVar("active_span_id")
```

Comment:
Ha. This is all we need eh? I didn't know that a single global ContextVar can be used to manage span_id's across both async tasks as well as threads. This definitely cleans up the manual shenanigans I resorted to using. Thanks!

Reply:
For threads, it also needs that `Thread` wrapper I added. For async it's automatic.

Reply:
Right. I saw that was really nice. So we should resort to using `llama_index.core.types.Thread` from now on (CC: @logan-markewich).